crates/starknet-types-core/Cargo.toml

```diff
@@ -1,13 +1,31 @@
 [package]
 name = "starknet-types-core"
-version = "0.0.2"
+version = "0.0.3"
 edition = "2021"
 license = "MIT"
 homepage = "https://github.com/starknet-io/types-rs"
 repository = "https://github.com/starknet-io/types-rs"
 categories = ["types", "math", "crypto"]
 keywords = ["stark", "zkp", "cairo"]
-description = "Starknet core types."
+description = "Core types representation for Starknet"
 readme = "README.md"
 
 [dependencies]
+bitvec = { version = "1.0.1", default-features = false }
+serde = { version = "1.0.163", optional = true, default-features = false }
+lambdaworks-math = { git = "https://github.com/lambdaclass/lambdaworks.git", rev = "f940e14ed17370d29fe129951448037d11b65ce8", default-features = false}
+
+
+arbitrary = { version = "1.3.0", optional = true, default-features = false }
+num-traits = { version = "0.2.16", default-features = false }
+
+[features]
+default = ["std", "serde", "curve"]
+curve = []
+std = []
+alloc = ["serde?/alloc"]
+arbitrary = ["std", "dep:arbitrary"]
```

Comment:
Honestly at this point if we do end up gating things behind feature flags, we should have a feature flag for `felt` as well.

Right now it wouldn't be very useful because everything depends on it, but we could imagine some types in the future that wouldn't.

Can you think of any type/feature that wouldn't depend on felt? I don't know the ecosystem well enough to think of one. If you can it would be wise to gate `Felt` as well (and enable that feature by default).

Reply:
For the plans we have in the starknet-core-types crate, almost everything depends on the Felt. I find it hard to imagine a user that would import this crate without the Felt type.
Do you have some examples?